src/cityreader/cityreader.py

```diff
@@ -60,12 +84,21 @@ def cityreader(cities=[]):
 
 # TODO Get latitude and longitude values from the user
 
+
 def cityreader_stretch(lat1, lon1, lat2, lon2, cities=[]):
-  # within will hold the cities that fall within the specified region
-  within = []
 
-  # TODO Ensure that the lat and lon valuse are all floats
-  # Go through each city and check to see if it falls within 
-  # the specified coordinates.
+    try:
+        lat_range = sorted([float(lat1), float(lat2)])
+        lon_range = sorted([float(lon1), float(lon2)])
+    except:
+        print(sys.exc_info())
+
+    within = [i for i in cities if i.lat >= lat_range[0]
+              and i.lat <= lat_range[1]
+              and i.lon >= lon_range[0]
+              and i.lon <= lon_range[1]]
+
+    return within
+
 
-  return within
```

Comment:
For the stretch, try asking the user for inputs once it gets to this point in the execution:
location1 = input(...stuff...).split(somehow)
location2 = " ... "

cityreader_stretch(location1[0]. location1[1], location2[0]......)